src/nomad_simulations/schema_packages/properties/band_structure.py

```diff
@@ -247,107 +98,23 @@ def extract_fermi_surface(self) -> FermiSurface | None:
         fermi_values = self.value[fermi_indices]
 
         # Store `FermiSurface` values
-        # ! This is wrong (!) the `value` should be the `KMesh.points`, not the `ElectronicEigenvalues.value`
+        # ! This is still conceptually wrong: `value` should be the KMesh.points,
+        # ! not the eigenvalues. Kept as-is until the FermiSurface property is final.
         fermi_surface = FermiSurface(
-            n_bands=self.n_bands,
+            n_levels=self.n_levels,
```

Comment:
So, we briefly discussed levels vs bands. I would say that bands are built out of levels due to Pauli exclusion in the reciprocal space. The DOS (from projecting out the band structure) instead matches best the energy diagrams of molecular levels. In theory, each band is continuous (along the k-space), but typically bound. When digitizing them, continuity becomes much harder to track, though.

1. If you agree, could you please add this semantics to our docs somewhere? Will look for a good reference (have it in mind, just blanking on the name).
2. Under this semantics, the FermiSurface is really just _bands_ that intersect the abstract Fermi energy level. It's those bands that we plot in real / reciprocal space. Levels are not really applicable here. I think @kubanmar agrees.


```suggestion
            n_bands=self.n_bands,
```

Note: in band structures I think that both apply, and one can derive bands from levels. This should be tackled in a follow-up issue. Could you open one and tag me pls?